Formula/b/beancount-language-server.rb

```diff
@@ -0,0 +1,35 @@
+class BeancountLanguageServer < Formula
+  desc "Language server for beancount files"
+  homepage "https://github.com/polarmutex/beancount-language-server"
+  url "https://github.com/polarmutex/beancount-language-server/archive/refs/tags/v1.3.4.tar.gz"
+  sha256 "d56f8d2d30f471b5f5011ffff9b55da9fecaec13f34a5f8f2ec0d6aa28cd3ded"
+  license "MIT"
+  head "https://github.com/polarmutex/beancount-language-server.git", branch: "main"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args(path: "crates/lsp")
+  end
+
+  test do
+    require "open3"
+
+    json = <<~JSON
+      {
+        "jsonrpc": "2.0",
+        "id": 1,
+        "method": "initialize",
+        "params": {
+          "rootUri": null,
+          "capabilities": {}
+        }
+      }
+    JSON
+
+    Open3.popen3("#{bin}/beancount-language-server", "--stdio") do |stdin, stdout|
+      stdin.write "Content-Length: #{json.size}\r\n\r\n#{json}"
+      assert_match(/^Content-Length: \d+/i, stdout.readline)
+    end
```

Comment:
Can't this use `pipe_output` for capturing the in and output?

Reply:
I tried using `pipe_output`, and the server panics on trying to write to a disconnected channel. I found the `open3` pattern in the formulae for other language servers, which I assume to be for the same reason based on explicit `sleep` calls before reading from stdout: https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-core+Open3.popen3+path%3A*-language-server.rb&type=code